### PR TITLE
Feature/change amount by text rb

### PR DESCRIPTION
**Pull request by lssalgado**
* Adding a dialog to input the desired amount of a drink. This dialog is shown when clicking the actual number of drinks;
* Renaming misleading function loadDrink to addNewDrink;
* Using Kotlin-like string concatenation.

### CODE REVIEW

File: app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt [hunk 1]
```diff
@@ -79,6 +80,9 @@ class DrinkFragment : BaseFragment(),DrinkMVPView, View.OnClickListener {
             R.id.btn_neg -> {
                 presenter.onNegDrinkSelected(mDrink!!)
             }
+            R.id.qnt_drink -> {
+                (activity as MainActivity).showChangeAmountDialog(mDrink!!)
```
Comment (eccard) [inline app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt]: Why not use drink fragment to show this dialog ?

File: app/src/main/java/com/r5k/contacerveja/ui/main/interactor/MainInteractor.kt [hunk 2]
```diff
@@ -66,4 +66,11 @@ class MainInteractor @Inject internal constructor(private val drinksRepoHelper:
         val drinkId = drinksRepoHelper.insertDrink(Drink(null, drinkName, null, 0, mOpenedBillId))
         Drink(drinkId,drinkName,null,0,mOpenedBillId)
     }
+
+
+    override suspend fun changeDrinkAmount(drink: Drink, amount: Int) = GlobalScope.async{
+        drink.qnt = amount
+        drinksRepoHelper.updateDrink(drink)
```
Comment (eccard) [inline app/src/main/java/com/r5k/contacerveja/ui/main/interactor/MainInteractor.kt]: We need to check the return of update query on database

File: app/src/main/java/com/r5k/contacerveja/ui/main/presenter/MainPresenter.kt [hunk 3]
```diff
@@ -99,9 +99,19 @@ class MainPresenter<V:MainMVPView, I : MainVMPInteractor> @Inject internal const
...
         }
     }
+
+    override fun changeDrinkAmount(drink: Drink, amount: Int) {
+        GlobalScope.launch(context = Dispatchers.Main) {
+            val updatedDrink = withContext(context = Dispatchers.IO) {
+                interactor!!.changeDrinkAmount(drink, amount).await()
+            }
+            Log.d(TAG,"changeDrinkAmount from=${drink.qnt} to=${updatedDrink.qnt}")
+            loadDrinks()
```
Comment (eccard) [inline app/src/main/java/com/r5k/contacerveja/ui/main/presenter/MainPresenter.kt]: Isn't necessary update all bills and drinks, we can update only the value of amount.

**Review by eccard — COMMENTED** [review]
I had improve some codes here #28.

**Review by eccard — APPROVED** [review]

File: app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt [hunk 4]
```diff
@@ -164,4 +168,36 @@ class DrinkFragment : BaseFragment(),DrinkMVPView, View.OnClickListener {
...
+                dialog.dismiss()
+            }
+        }
+
+        builder.setNegativeButton(android.R.string.cancel) { dialog, p1 ->
+            dialog.cancel()
+        }
+
+        builder.show()
+    }
```
Comment (leomsaippa) [inline app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt:201]: This method should be in a DialogHelper

Comment (eccard) [reply app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt:201]: There are a lot of duplicated code about dialog at here and in MainActivity, let's create this helper in another branch.

Comment (eccard) [reply app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt:201]: I have created this issue #29

**Review by leomsaippa — APPROVED** [review]

**Review by m4theuslim4 — APPROVED** [review]

**Pull request merged**